aurora/core.py

```diff
@@ -169,29 +178,42 @@ def load_dict(self, aurora_dict):
     def setup_grids(self):
         """Method to set up radial and temporal grids given namelist inputs.
         """
-        # Get r_V to rho_pol mapping
-        rho_pol, _rvol = grids_utils.get_rhopol_rvol_mapping(self.geqdsk)
-
-        rvol_lcfs = interp1d(rho_pol, _rvol)(1.0)
-        self.rvol_lcfs = self.namelist["rvol_lcfs"] = np.round(
-            rvol_lcfs, 3
-        )  # set limit on accuracy
+        if 'rvol_lcfs' in self.namelist:
+            # separatrix location explicitly given by user
+            self.rvol_lcfs = self.namelist['rvol_lcfs']
+
+        elif hasattr(self, 'geqdsk'):
+            # Get r_V to rho_pol mapping
+            rho_pol, _rvol = grids_utils.get_rhopol_rvol_mapping(self.geqdsk)
+            rvol_lcfs = interp1d(rho_pol, _rvol)(1.0)
+            self.rvol_lcfs = self.namelist["rvol_lcfs"] = np.round(
+                rvol_lcfs, 3
+            )  # set limit on accuracy
 
+        else:
+            raise ValueError('Could not identify rvol_lcfs. Either provide this in the namelist or provide a geqdsk equilibrium')
+            
         # create radial grid
         grid_params = grids_utils.create_radial_grid(self.namelist, plot=False)
         self.rvol_grid, self.pro_grid, self.qpr_grid, self.prox_param = grid_params
 
-        # get rho_poloidal grid corresponding to aurora internal (rvol) grid
-        self.rhop_grid = interp1d(_rvol, rho_pol, fill_value="extrapolate")(
-            self.rvol_grid
-        )
-        self.rhop_grid[0] = 0.0  # enforce on axis
+        if hasattr(self, 'geqdsk') and len(self.geqdsk):
+            # get rho_poloidal grid corresponding to aurora internal (rvol) grid
+            self.rhop_grid = interp1d(_rvol, rho_pol, fill_value="extrapolate")(
+                self.rvol_grid
+            )
+            self.rhop_grid[0] = 0.0  # enforce on axis
 
-        # Save R on LFS and HFS
-        self.Rhfs, self.Rlfs = grids_utils.get_HFS_LFS(
-            self.geqdsk, rho_pol=self.rhop_grid
-        )
+            # Save R on LFS and HFS
+            #self.Rhfs, self.Rlfs = grids_utils.get_HFS_LFS(
+            #    self.geqdsk, rho_pol=self.rhop_grid
+            #)
+            
+        else:
+            # use rho_vol = rvol/rvol_lcfs
+            self.rhop_grid = self.rvol_grid / self.rvol_lcfs
```

Comment:
in principle, we could remove rhop entirely from aurora and use rhov instead. It might be confusing to call it rhop even if it is something else.

Reply:
Yes, one could use `rhov` or even just a general `rho` (which I don't like much though, because it leaves ambiguity. However, the issue is that this would not be a backwards compatible change... and I'm not sure that we want to make a big thing out of this change, which is only important for W7-X at the moment... so I'm tempted to just leave it like this and just "alert" W7-X folks that Aurora refers to `rhop`, but in W7-X's case this is just a rhov. (FYI, W7-X people don't talk about `rhov` either - it's just called `reff` for stellarator research, and it's the standard radial coordinate)